### PR TITLE
[FEATURE] Require IDs for ExpectationSuite and Expectations

### PR DESCRIPTION
**Pull request by joshua-stauffer**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/data_context/store/expectations_store.py
Comment (tyler-hoffman) [inline great_expectations/data_context/store/expectations_store.py:171]: Still looking at the PR, but I think it's a problem to assign `ge_cloud_id` when we are not in cloud mode.

File: great_expectations/data_context/store/expectations_store.py
Comment (tyler-hoffman) [inline great_expectations/data_context/store/expectations_store.py:175]: How does mypy let us get here given the method signature? Shouldn't require actual domain objects rather than dicts at this point? If we really have a use case for dicts (which I'm assuming we do, given you're handling it), I think we should have a separate method for each (and in the suite one, we should use dot notation for accessing values)

File: great_expectations/data_context/store/expectations_store.py
Comment (joshua-stauffer) [reply great_expectations/data_context/store/expectations_store.py:171]: do you know where this is a problem? i was wondering if someone has pinned logic to `ge_cloud_id`, but haven't run across an example of it. it'll be renamed to just `id` in a followup PR.

File: great_expectations/data_context/store/expectations_store.py
Comment (tyler-hoffman) [inline great_expectations/data_context/store/expectations_store.py]: Feels off to even call this method if we are in cloud method. I'd put this check outside of this method.

File: great_expectations/data_context/store/gx_cloud_store_backend.py
Comment (tyler-hoffman) [inline great_expectations/data_context/store/gx_cloud_store_backend.py:338]: wat

File: great_expectations/data_context/store/gx_cloud_store_backend.py
Comment (tyler-hoffman) [inline great_expectations/data_context/store/gx_cloud_store_backend.py:292]: This signature stresses me out. Were we actually using the return value before? Looks like we only ever returned `True`, making the return value useless IIUC. Can we just change it to `GXCloudResourceRef`?

File: tests/core/test_expectation_suite.py
Comment (tyler-hoffman) [inline tests/core/test_expectation_suite.py:383]: I think it's probably worthwhile to assert that these don't have uuids before being added to the suite as well so that we actually know that the suite is handling it.

**Review by tyler-hoffman — CHANGES_REQUESTED** [review]
High level seems about right. I have concerns about reusing the ge_cloud_id. We should either rename it `id` (I'm fine if that's in a subsequent PR to avoid blowing up the scope of this), or we should have both an `id` and the separate `ge_cloud_id` (I thought that was a pattern in some of our code?)

File: great_expectations/data_context/store/expectations_store.py
Comment (joshua-stauffer) [reply great_expectations/data_context/store/expectations_store.py]: thanks, reasonable. updated in [4b1a9e6](https://github.com/great-expectations/great_expectations/pull/9105/commits/4b1a9e61cf0ac5d50c5f9552c31bc66de6003c98)

File: great_expectations/data_context/store/gx_cloud_store_backend.py
Comment (joshua-stauffer) [reply great_expectations/data_context/store/gx_cloud_store_backend.py:338]: yeah. this method is used by all store implementations, and other `update` calls (like Checkpoint) do return a boolean. fixing the store implementation is explicitly out of scope, but let's improve this soon.

File: great_expectations/data_context/store/gx_cloud_store_backend.py
Comment (joshua-stauffer) [reply great_expectations/data_context/store/gx_cloud_store_backend.py:292]: true - unfortunately other `update` endpoints still return bools

File: great_expectations/data_context/store/expectations_store.py
Comment (joshua-stauffer) [reply great_expectations/data_context/store/expectations_store.py:175]: the `migrator` code is passing a`dict` here. I think that's probably wrong, and would like to remove that code, but that feels out of scope for this work, and this fork is required to get tests to pass.

File: great_expectations/data_context/store/expectations_store.py
Comment (tyler-hoffman) [reply great_expectations/data_context/store/expectations_store.py:171]: I think it's just a really misleading name at this point. Let's follow up and rename.

File: great_expectations/data_context/store/expectations_store.py
Comment (tyler-hoffman) [reply great_expectations/data_context/store/expectations_store.py:175]: I would like the signature to read suite: `dict | ExpectationSuite` before this merges

File: great_expectations/data_context/store/gx_cloud_store_backend.py
Comment (tyler-hoffman) [reply great_expectations/data_context/store/gx_cloud_store_backend.py:338]: Can you add a "why" comment on this!

**Review by tyler-hoffman — APPROVED** [review]
Approved pending convos being resolved.

File: great_expectations/data_context/store/gx_cloud_store_backend.py
Comment (joshua-stauffer) [reply great_expectations/data_context/store/gx_cloud_store_backend.py:338]: even better - updated this to use the returned HTTP code instead of exceptions. It does require that the backend use the correct code, but is definitely easier to reason about

File: great_expectations/data_context/store/expectations_store.py
Comment (joshua-stauffer) [reply great_expectations/data_context/store/expectations_store.py:175]: unfortunately that change causes other typing issues. mypy seems convinced that the only thing ever passed to this object is a dict, even though the `CloudMigrator` is passing a dict. For now I'll leave this signature in place, but I'll add a ticket to remove the migrator as a follow up, and change this code to use dot notation. It would need to change at the latest when we migrate the suite object to pydantic, since bracket syntax will no longer work.

**Pull request merged**
